functions/sync_images/main.py

```diff
@@ -0,0 +1,148 @@
+import json
+import logging
+
+from config import (
+    IMAGE_STORE_BUCKET,
+    TOPIC_NAME,
+    ENTRY_FILEPATH_PREFIX
+)
+
+from attachment_service import AttachmentService
+from form_object import Form
+from google.cloud import storage
+from publish_service import PublishService
+from requests_retry_session import get_requests_session
+
+logging.basicConfig(level=logging.INFO)
+
+
+def handler(request):
+    """
+    This cloud function will attempt to correct ArcGIS features
+    when needed. The repair mainly focuses on checking the attachments,
+    and download them when missing. When sending the form to
+    the ArcGIS interface it also causes the interface to remap all fields.
+    This is desirable since it updates the ArcGIS features to their
+    desired state.
+
+    :param request: The request to this cloud function.
+    :type request: flask.Request
+
+    :return: The result of this cloud function., An HTTP status code.
+    :rtype: str, int
+    """
+
+    # Initializing components
+    arguments = get_request_arguments(request)
+
+    # Skips download of attachments.
+    skip_download = arguments.get("skip_download", False)
+
+    # Suffix of the form bucket path.
+    form_storage_suffix = arguments.get("form_storage_suffix", "")
+
+    # Update arcgis entry even when no attachment was downloaded.
+    force_arcgis_update = arguments.get("force_arcgis_update", False)
+
+    # Retry options
+    request_retry_options = arguments.get("request_retry_options", {})
+
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(IMAGE_STORE_BUCKET)
+
+    attachment_service = AttachmentService(storage_client, get_requests_session(**request_retry_options))
```

Comment:
abstraction is recommended. why not hide get_requests_session within the functions, so users of that functions don't have to worry about it when calling the function.